src/components/tooltip/Marker.js

```diff
@@ -572,49 +568,60 @@ function getMarkerDetails(
             </TooltipDetail>
           </TooltipDetails>
         );
+        break;
       }
       case 'Invalidation': {
-        return (
+        tooltipDetails = (
           <TooltipDetails>
             <TooltipDetail label="URL">{data.url}</TooltipDetail>
             <TooltipDetail label="Line">{data.line}</TooltipDetail>
           </TooltipDetails>
         );
+        break;
       }
       case 'Network': {
-        return <TooltipNetworkMarker payload={data} zeroAt={zeroAt} />;
+        tooltipDetails = (
+          <TooltipNetworkMarker payload={data} zeroAt={zeroAt} />
+        );
+        break;
       }
       case 'Styles': {
-        return (
-          <>
-            <TooltipDetails>
-              <TooltipDetail label="Elements traversed">
-                {data.elementsTraversed}
-              </TooltipDetail>
-              <TooltipDetail label="Elements styled">
-                {data.elementsStyled}
-              </TooltipDetail>
-              <TooltipDetail label="Elements matched">
-                {data.elementsMatched}
-              </TooltipDetail>
-              <TooltipDetail label="Styles shared">
-                {data.stylesShared}
-              </TooltipDetail>
-              <TooltipDetail label="Styles reused">
-                {data.stylesReused}
-              </TooltipDetail>
-            </TooltipDetails>
-            {_markerBacktrace(marker, data, thread, implementationFilter)}
-          </>
+        tooltipDetails = (
+          <TooltipDetails>
+            <TooltipDetail label="Elements traversed">
+              {data.elementsTraversed}
+            </TooltipDetail>
+            <TooltipDetail label="Elements styled">
+              {data.elementsStyled}
+            </TooltipDetail>
+            <TooltipDetail label="Elements matched">
+              {data.elementsMatched}
+            </TooltipDetail>
+            <TooltipDetail label="Styles shared">
+              {data.stylesShared}
+            </TooltipDetail>
+            <TooltipDetail label="Styles reused">
+              {data.stylesReused}
+            </TooltipDetail>
+          </TooltipDetails>
         );
+        break;
       }
       case 'tracing': {
-        return _markerBacktrace(marker, data, thread, implementationFilter);
+        tooltipDetails = _getTracingMarkerDetails(marker);
+        break;
       }
       default:
+        return null;
     }
   }
-  return null;
+
+  return (
+    <>
+      {tooltipDetails}
+      {_markerBacktrace(marker, thread, implementationFilter)}
+    </>
```

Comment:
I think we still need to return `null` if there's no details and no backtrace. Because otherwise we have a wrong rendering for those markers, for example jank markers, where we have a separator when we shouldn't have:
![image](https://user-images.githubusercontent.com/454175/60181433-79209800-9822-11e9-8494-c32fc6231a36.png)


The reason is that now we always return something here (even if it's an empty fragment) so the conditional class in 
```js
         <div className={classNames({ tooltipHeader: details })}>
```
is always added.

Reply:
Ah, yeah. Returning null now.